src/main/java/com/yunhalee/walkerholic/activity/service/ActivityService.java

```diff
@@ -1,98 +1,88 @@
 package com.yunhalee.walkerholic.activity.service;
 
-import com.yunhalee.walkerholic.util.FileUploadUtils;
-import com.yunhalee.walkerholic.activity.dto.ActivityCreateDTO;
-import com.yunhalee.walkerholic.activity.dto.ActivityDTO;
+import com.yunhalee.walkerholic.activity.dto.ActivityRequest;
+import com.yunhalee.walkerholic.activity.exception.ActivityNotFoundException;
+import com.yunhalee.walkerholic.common.service.S3ImageUploader;
+import com.yunhalee.walkerholic.activity.dto.ActivityResponse;
+import com.yunhalee.walkerholic.activity.dto.ActivityDetailResponse;
 import com.yunhalee.walkerholic.activity.domain.Activity;
 import com.yunhalee.walkerholic.activity.domain.ActivityRepository;
-import lombok.RequiredArgsConstructor;
+import java.io.IOException;
+import java.util.stream.Collectors;
 import org.springframework.stereotype.Service;
-import org.springframework.util.StringUtils;
+import org.springframework.transaction.annotation.Transactional;
 import org.springframework.web.multipart.MultipartFile;
 
-import java.io.IOException;
-import java.util.ArrayList;
 import java.util.List;
 
 @Service
-@RequiredArgsConstructor
+@Transactional(readOnly = true)
```

Comment:
```suggestion
@Transactional
```
readOnly 인 부분에서만 옵션을 추가하는 것은 어떨까요?

Reply:
전체적으로 @Transactional을 주고 해당부분만 readOnly옵션을 적용하도록 변경하였습니다!